### ISSUE TITLE
DATALOGGER IOT - 9DOF DOES NOT RECOGNIZE SAM-M8Q GPS BOARD

### ISSUE THREAD

**Issue by csqrd** [sparkfun/SparkFun_DataLogger_IoT_9DoF#1]
It appears that the datalogger does not recognize the M8Q as it does not show up as a device along with the other 3 built-in devices. I updated the datalogger to firmware version 01.00.02 which did not resolve the issue. I verified that the SAM-M8Q board is functioning properly by using the U-Blox u-center software and a SparkFun u-blox GNSS Arduino example sketch. The I2C address is 0x42. Any help would be appreciated.

 I did notice "[W] GNSS::isConnected no traffic seen (first attempt)" message but would expect the module to show up in the device list? I assumed a GPS fix was established by the 1 per second flashing "PPM" LED but will need to use a arduino sketch to confirm.

Here is the datalogger output:

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0
E (1096) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
E (1097) vfs_fat_sdmmc: sdmmc_card_init failed (0x107).
[E] Startup of the SD card failed.
[W] Error initializing the SD Card. Is an SD card installed on the board?

SparkFun DataLogger IoT - 9DOF (c) 2023 SparkFun Electronics
Version: 01.00.02 - build 000098

[W] GNSS::isConnected no traffic seen (first attempt)
Restoring System Settings ...restored from ESP32 Preferences
Connecting to WiFi...Connected to gnet
NTP startup [5 secs].......enabled
Loading devices ... 3 devices detected
MAX17048 - MAX17048 LiPo Battery Fuel Gauge {qwiic}
ISM330 - ISM330 Inertial Measurement Unit {SPI}
MMC5983 - MMC5983 Magnetometer {SPI}

Time: 2023-06-12T14:40:59
Uptime: 0 days, 00:00:16.246
External Time Source: NTP Client
Board Name: SparkFun DataLogger IoT - 9DoF
Board ID: SFD1D085B886E694
SD card not available
[I] WiFi - Connected SSID: gnet IP Address: 192.168.xxx.xxx Signal: Excellent
[I] Battery - Not Connected
[I] System Deep Sleep: disabled
Sleep Interval: 0 seconds
Wake Interval: 0 seconds

[I] Logging Interval (ms): 15000
[I] Serial Output: CSV Format
Baud Rate: 115200

**Comment by PaulZC** [comment]
Hi @csqrd ,

Apologies for the inconvenience. I can replicate this issue.

TLDR: I think the root cause is a flaw in the DataLogger firmware driver for the u-blox GNSS. I think it is caused by the way the GNSS module is detected - by looking for messages coming from the module. This works the first time - when the module is outputting NMEA messages by default. But the driver then disables the NMEA messages, saves the port configuration, and only then enables the UBX-NAV-PVT message. When the DataLogger firmware restarts, there are no NMEA messages (because they have been disabled), and no PVT messages (because the PVT message configuration is not saved).

I will confirm this, talk to my colleagues and get this fixed in the next release.

Very best wishes,
Paul

**Comment by csqrd** [comment]
Thanks for the explanation.  Excited for a fix as I have several projects planed.

**Comment by PaulZC** [comment]
Hi @csqrd ,

Thank you for your patience. We have a working fix for this. A new version of the firmware will be available shortly.

Best wishes,
Paul

**Comment by DragonFlameBoy** [comment]
Hello 

This is my first post so apologies if this is not the correct posting approach. 

I have the same issue with the **GPS-18037 MAX-M10S** its not recognised except for the fist line - just as with the SAM-M8Q so I assume its the same issue?

SparkFun DataLogger IoT - 9DOF   (c) 2023 SparkFun Electronics
Version: 01.00.02  - build 000098
[W] GNSS::isConnected no traffic seen (first attempt)
[I] Restoring System Settings ...restored from ESP32 Preferences
[I] Connecting to WiFi...Connected to Vulcan
[I] NTP startup [5 secs].......enabled
[I] Loading devices ... 3 devices detected
    MAX17048              - MAX17048 LiPo Battery Fuel Gauge  {qwiic}
    ISM330                - ISM330 Inertial Measurement Unit  {SPI}
    MMC5983               - MMC5983 Magnetometer  {SPI}

Kind regards

Chris

**Comment by gigapod** [comment]
Hi - 

I believe this  is fixed and will be in the next build/release. Will check with Engineer on this. 

If everything goes well, we should have the current fix for this issue wrapped on on Friday and published. 

Thanks for your patience! 

-Kirk

**Comment by gigapod** [comment]
Firmware version 1.00.03 was just published and includes the fix for this issue. 

The update can be applied via the menu system on the DataLogger - See update  instructions [here](https://docs.sparkfun.com/SparkFun_DataLogger/configuration/#advanced-system-update)

**Comment by DragonFlameBoy** [comment]
I can confirm Firmware version 1.00.03 resolves the issue with GPS-18037 MAX-M10S.  I now get the following additional information logged on the SparkFun DataLogger IoT - 9DOF:

GNSS.Year
GNSS.Month
GNSS.Day
GNSS.Hour
GNSS.Minute
GNSS.Second
GNSS.Latitude (deg)
GNSS.Longitude (deg)
GNSS.Altitude (m)
GNSS.Altitude MSL (m)
GNSS.SIV
GNSS.Fix Type
GNSS.Carrier Solution
GNSS.Ground Speed (m/s)
GNSS.Heading (deg)
GNSS.PDOP
GNSS.Horizontal Accuracy Estima
GNSS.Vertical Accuracy Estimate
GNSS.TOW (ms)
GNSS.ISO8601
GNSS.YYYYMMDD
GNSS.YYYYDDMM
GNSS.DDMMYYYY
GNSS.HHMMSS
GNSS.Fix Type (String)
GNSS.Carrier Solution (String)

Many thanks to **csqrd** for reporting this and SparkFun for resolving this.

Chris

**Comment by gigapod** [comment]
@DragonFlameBoy  - thank you for verifying the fix  is working - always helpful to get feedback.

If you have any other issues/inputs - don't hesitate to file an issue here, or on the repo we use for the firmware (https://github.com/sparkfun/SparkFun_DataLogger/tree/main).

-Kirk

**Comment by csqrd** [comment]
I can verify that the M8Q works as well.  
Thanks!
